codesquad-members-2021/java-chess
# [yeon] 미션2: 체스판 생성

안녕하세요! yeon입니다. 미션 2단계 체스판 생성 구현하고 리뷰 요청합니다 😊
아래에 각 클래스별 필드와 메서드를 설명드리겠습니다.

 1. 체스판에 해당하는 **Board 클래스**
- pawns : Pawn 타입의 객체만 담을 수 있는 ArrayList를 만들었습니다.
- add() : 인자로 받은 pawn을 ArrayList에 담는 기능을 합니다.
- size() : ArrayList의 크기를 반환합니다.
- findPawn() : 인자로 받은 숫자의 위치에 해당하는 Pawn 객체를 반환합니다.

2. **BoardTest 클래스**
- Pawn 타입의 전역변수 white, black 를 추가했습니다.
- 체스판에 해당하는 전역변수 board 를 추가했습니다.
- check() : 예상되는 크기(size), 해당 Pawn 타입 객체의 반환이 예측되는 index값 그리고 Pawn타입의 객체를 인자로 넣어서
객체를 board에 추가한 뒤, board 메소드의 반환값과 같은지 비교하는 메소드입니다.
- create() : check()메서드를 넣어서 테스트합니다.

3. **Pawn 클래스**
- 기본생성자 추가: 기본생성자로 생성된 객체는 white 색을 갖도록 합니다.
- WHITE, BLACK: white, black 의 string 값을 갖는 상수를 추가했습니다.

4. **PawnTest 클래스**
- create_기본생성자() 메서드 추가: 기본생성자로 객체 생성시 객체의 color가 white인지 확인해주는 코드를 작성했습니다.


여기까지 미션2를 구현했습니다.
귀한 시간 내서 코드리뷰 해주셔서 감사합니다!
